Code/GraphMol/Wrap/MolOps.cpp

```diff
@@ -305,9 +305,15 @@ void addRecursiveQueriesHelper(ROMol &mol, python::dict replDict,
   addRecursiveQueries(mol, replacements, propName);
 }
 
-ROMol *addHs(const ROMol &orig, bool explicitOnly = false,
-             bool addCoords = false) {
-  return MolOps::addHs(orig, explicitOnly, addCoords);
+ROMol *addHs(const ROMol &orig, bool explicitOnly, bool addCoords,
+             python::object onlyOnAtoms) {
+  std::vector<unsigned int> *onlyOn = NULL;
+  if (onlyOnAtoms) {
+    onlyOn = pythonObjectToVect(onlyOnAtoms, orig.getNumAtoms());
+  }
+  ROMol *res = MolOps::addHs(orig, explicitOnly, addCoords, onlyOn);
```

Comment:
In this case why not have the invariant inside the c++ code so we don't need it in the wrappers?

This would solve the issue for C++/Java/C# as well.  It might muck with the c++ logic though as you would need something like

PRECONDITION(std::min( onlyOnAtoms ) < mol.GetNumAtoms() )